sync/task_pool.go

```diff
@@ -84,14 +84,21 @@ func WithTaskPoolTaskQueueNumber(number int) TaskPoolOption {
 /////////////////////////////////////////
 
 // task t
-type task func()
+type runnable func() (interface{}, error)
+
+type consum func(interface{}, error)
+
+type task struct {
+	run          runnable
```

Comment:
you can define a function as a task , in which calling runnable first and callback the result:
```
pool.AddTask(func(){
    callback(runnable())
})
```

Reply:
good idea.